lib/countly.js

```diff
@@ -3316,6 +3368,53 @@
         return ob;
     }
 
+    /**
+     * Truncates an object's key/value pairs to a certain length
+     * @param {Object} obj - original object to be truncated
+     * @param {Number} keyLimit - limit for key length
+     * @param {Number} valueLimit - limit for value length
+     * @returns {Object} - the new truncated object
+     */
+    function truncateObject(obj, keyLimit, valueLimit) {
+        var ob = {};
+        if (Object.keys(obj).length > this.maxSegmentationValues) {
+            var resizedObj = {};
+            for (var e in obj) {
+                resizedObj[e] = obj[e];
+            }
+            obj = resizedObj;
+        }
+        for (var key in obj) {
+            var newKey = key;
+            var newValue = obj[key];
+            if (key.length > keyLimit) {
+                newKey = key.substring(0, keyLimit);
+                this.log("Key: " + key + " is longer than accepted length. It will be truncated.");
+            }
+            if (obj[key].length > valueLimit && (typeof obj[key] === 'string' || typeof obj[key] === 'number')) {
```

Comment:
would probably need similar early type checking as described for "truncateSingleValue"